src/main/java/seedu/address/logic/commands/PayCommand.java

```diff
@@ -0,0 +1,146 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_HOUR;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_STUDENTS;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.student.Address;
+import seedu.address.model.student.Email;
+import seedu.address.model.student.Name;
+import seedu.address.model.student.OwedAmount;
+import seedu.address.model.student.PaidAmount;
+import seedu.address.model.student.Phone;
+import seedu.address.model.student.Rate;
+import seedu.address.model.student.Schedule;
+import seedu.address.model.student.Student;
+import seedu.address.model.student.Subject;
+
+/**
+ * Updates the amount of tuition fee paid by an existing student in the address book.
+ */
+public class PayCommand extends Command {
+
+    public static final String COMMAND_WORD = "pay";
+    public static final String COMMAND_WORD_RANDOM_CASE = "PaY";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Updates the payment made by the student identified "
+            + "by the index number used in the displayed Student list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_HOUR + "HOURS PAID\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_HOUR + "3 ";
+
+    public static final String MESSAGE_EDIT_STUDENT_SUCCESS = "%1$s paid $%2$.2f";
+    public static final String MESSAGE_DUPLICATE_STUDENT = "This Student already exists in the address book.";
```

Comment:
This message might be redundant as it seems to not be used anywhere

Reply:
Thanks for catching this! Ive removed it